js/background.js

```diff
@@ -483,40 +499,45 @@
         });
     }
 
-    function onMessageReceived(ev) {
-        var data = ev.data;
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            var profileKey = data.message.profileKey.toArrayBuffer();
-            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
-              return sender.setProfileKey(profileKey).then(ev.confirm);
-            });
-        }
-        var message = initIncomingMessage(data);
-
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
+  /* eslint-enable */
+  /* jshint ignore:start */
+  async function onMessageReceived(ev) {
+    const { data } = ev;
+
+    // eslint-disable-next-line no-bitwise
+    if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
+      const profileKey = data.message.profileKey.toArrayBuffer();
+      const sender = ConversationController.getOrCreateAndWait(data.source, 'private');
```

Comment:
Should this line have an await after the `=`?

Reply:
Good catch 🎣 ! Talked to Matthew about how to test this and he said that regular avatar change does not trigger profile key update as it reuses the existing one. Do rotate it, we’d need to re-register an account or build special functionality into one of the clients. Instead, I temporarily set `if` condition to `true` and tested that existing messages with profile keys flow through this code without error. See: 10a3f2f56f403338d2dc40306a1795fe9f95f6ed